File: src/test/java/com/davidgracia/euclideantsp/test/Euclidean2DTSPInstances.java
package com.davidgracia.euclideantsp.test;

import com.davidgracia.euclideantsp.Euclidean2DTSPInstance;
import org.locationtech.jts.geom.Coordinate;

import java.util.Arrays;

class Euclidean2DTSPInstances {
    static final Euclidean2DTSPInstance INSTANCE_1 = new Euclidean2DTSPInstance(
            Arrays.asList(
                    new Coordinate(0, 0),
                    new Coordinate(0, 1),
                    new Coordinate(1, 0))
    );

    static final Euclidean2DTSPInstance INSTANCE_2 = new Euclidean2DTSPInstance(
            Arrays.asList(
                    new Coordinate(0, 0),
                    new Coordinate(1, 0),
                    new Coordinate(2, 0),
                    new Coordinate(1, 2))
    );


    static final Euclidean2DTSPInstance INSTANCE_3 = new Euclidean2DTSPInstance(
            Arrays.asList(
                    new Coordinate(1, 3),
                    new Coordinate(1, 0),
                    new Coordinate(0, 2),
                    new Coordinate(2, 4),
                    new Coordinate(3, 3))
    );

    static final Euclidean2DTSPInstance INSTANCE_4 = new Euclidean2DTSPInstance(
            Arrays.asList(
                    new Coordinate(3, 0),
                    new Coordinate(4, 5),
                    new Coordinate(3, 3),
                    new Coordinate(0, 4),
                    new Coordinate(10, 4),
                    new Coordinate(4, 6))
    );

    static final Euclidean2DTSPInstance INSTANCE_5 = new Euclidean2DTSPInstance(
            Arrays.asList(
                    new Coordinate(0, 0),
                    new Coordinate(0, 4),
                    new Coordinate(7, 0),
                    new Coordinate(7, 4),
                    new Coordinate(3, 2),
                    new Coordinate(4, 2))
    );

    static final Euclidean2DTSPInstance INSTANCE_6 = new Euclidean2DTSPInstance(
            Arrays.asList(
                    new Coordinate(0, 0),
                    new Coordinate(0, 4),
                    new Coordinate(7, 0),
                    new Coordinate(7, 4),
                    new Coordinate(1, 2),
                    new Coordinate(6, 2))
    );

}

File: src/test/kotlin/com/github/dgraciac/euclideantsp/Euclidean2DTSPInstances.kt
package com.github.dgraciac.euclideantsp

import org.locationtech.jts.geom.Coordinate

internal object Euclidean2DTSPInstances {
    val INSTANCE_1 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(0.0, 0.0),
            Coordinate(0.0, 1.0),
            Coordinate(1.0, 0.0)
        )
    )
    val INSTANCE_2 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(0.0, 0.0),
            Coordinate(1.0, 0.0),
            Coordinate(2.0, 0.0),
            Coordinate(1.0, 2.0)
        )
    )
    val INSTANCE_3 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(1.0, 3.0),
            Coordinate(1.0, 0.0),
            Coordinate(0.0, 2.0),
            Coordinate(2.0, 4.0),
            Coordinate(3.0, 3.0)
        )
    )
    val INSTANCE_4 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(3.0, 0.0),
            Coordinate(4.0, 5.0),
            Coordinate(3.0, 3.0),
            Coordinate(0.0, 4.0),
            Coordinate(10.0, 4.0),
            Coordinate(4.0, 6.0)
        )
    )
    val INSTANCE_5 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(0.0, 0.0),
            Coordinate(0.0, 4.0),
            Coordinate(7.0, 0.0),
            Coordinate(7.0, 4.0),
            Coordinate(3.0, 2.0),
            Coordinate(4.0, 2.0)
        )
    )
    val INSTANCE_6 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(0.0, 0.0),
            Coordinate(0.0, 4.0),
            Coordinate(7.0, 0.0),
            Coordinate(7.0, 4.0),
            Coordinate(1.0, 2.0),
            Coordinate(6.0, 2.0)
        )
    )
    val INSTANCE_7 = Euclidean2DTSPInstance(
        listOf(
            Coordinate(0.0, 0.0),
            Coordinate(18.0, 0.0),
            Coordinate(9.0, 18.0),
            Coordinate(2.0, 1.0),
            Coordinate(3.0, 1.0),
            Coordinate(3.0, 2.0)
        )
    )
}
